v2/public/localscripts/scatter.js

```diff
@@ -114,15 +116,25 @@ function scatter(selector, xdata, ydata, options) {
     svg.append("line")
         .attr({x1: width, y1: 0, x2: width, y2: height, "class": "y axis"});
 
-    var pData = d3.zip(xdata, ydata);
-    svg.selectAll(".point")
-        .data(pData)
-        .enter().append("circle")
+    var pData = d3.zip(xdata, ydata, labels);
+
+    var node = svg.selectAll(".point")
+                .data(pData)
+                .enter()
+                .append("g");
+    console.log(node);
+
+    node.append("circle")
         .attr("class", "point")
         .attr("cx", function(p) {return x(p[0]);})
         .attr("cy", function(p) {return y(p[1]);})
         .attr("r", function(p) {return options.radius;});
 
+    node.append("text")
+      .attr("x", function(p) { return x(p[0]) - 4;})
```

Comment:
We shouldn't be doing a hard-coded x-shift here to put the text centered above the point (for example, the label for question "11" is not correctly centered at the moment). Instead we should have `"x"` be exactly `x(p[0])` and we should change the text horizontal alignment to be centered.